wagtail/admin/views/pages/listing.py

```diff
@@ -20,13 +28,33 @@
     PageTable,
     PageTitleColumn,
 )
-from wagtail.admin.views.generic.base import BaseListingView
-from wagtail.admin.views.generic.permissions import PermissionCheckedMixin
-from wagtail.models import Page
+from wagtail.admin.views import generic
+from wagtail.models import Page, get_page_models
 from wagtail.permissions import page_permission_policy
 
 
-class BaseIndexView(PermissionCheckedMixin, BaseListingView):
+def get_content_types_for_filter():
+    models = [model.__name__.lower() for model in get_page_models()]
+    return ContentType.objects.filter(model__in=models).order_by("model")
```

Comment:
I just noticed that if the project has two similarly named models in different apps, e.g. a `Post` (non-page) model in app A and a `Post` page model in app B, the filter could return both. We should also filter by the app label.

I think this is easier with the built-in `ContentType.objects.get_for_models`, which will also make use of the manager's cache when possible. I'll add this as I merge.

Not adding a test yet though, as that would require adding a model without the `Page` suffix in the test models, and I don't have time for it now 😂

Edit: it's harder than I thought, since the filter expects a queryset. I think I'll leave this to be handled in a future PR to unblock things for now.